include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -70,17 +74,34 @@ namespace LinearAlgebra
     Vector<Number>::Vector(const Teuchos::RCP<VectorType> V)
       : Subscriptor()
       , vector(V)
-    {}
+    {
+      if (vector->getMap()->isOneToOne())
+        {
+          compressed = true;
+          locally_owned_map =
+            Teuchos::rcp_const_cast<MapType>(vector->getMap());
+        }
+      else
+        {
+          compressed = false;
+          locally_relevant_map =
+            Teuchos::rcp_const_cast<MapType>(vector->getMap());
+        }
+    }
 
 
 
     template <typename Number>
     Vector<Number>::Vector(const IndexSet &parallel_partitioner,
                            const MPI_Comm  communicator)
       : Subscriptor()
-      , vector(Utilities::Trilinos::internal::make_rcp<VectorType>(
-          parallel_partitioner.make_tpetra_map_rcp(communicator, false)))
-    {}
+    {
+      compressed = true;
+      locally_owned_map =
+        parallel_partitioner.make_tpetra_map_rcp(communicator, false);
+      vector =
+        Utilities::Trilinos::internal::make_rcp<VectorType>(locally_owned_map);
```

Comment:
These could all go into the member initializer list.

Reply:
I moved the members into the initializer list.